internal/services/network/ip_group_cidr_resource.go

```diff
@@ -136,10 +136,11 @@ func resourceIpGroupCidrRead(d *pluginsdk.ResourceData, meta interface{}) error
 		if resp.Model.Properties == nil {
 			return fmt.Errorf("retrieving %s: `properties` was nil", ipGroupId)
 		}
-		if !utils.SliceContainsValue(*resp.Model.Properties.IPAddresses, cidr) {
-			d.SetId("")
-			return nil
-		}
+	}
+
+	if !utils.SliceContainsValue(*resp.Model.Properties.IPAddresses, cidr) {
```

Comment:
Since `IPAddresses` is a pointer this also wants to be nil checked before dereferencing, alternatively we can use `pointer.From` to do that for us 
```suggestion
	if !utils.SliceContainsValue(pointer.From(resp.Model.Properties.IPAddresses), cidr) {
```

Reply:
thanks for the suggestion 👍